### PR TITLE
Test For command falco -i (ignore default events) 

### PR DESCRIPTION
**Pull request by Rohith-Raju**
This issue is related to : #7, 
Checks of task
- [ ] i 

Implementation: 
I moved all the ignored events to the `events.json` file. The reason behind this was that if we were to store all the events in the test, it would look dirty.
Read the `event.json` file, loop over the events, and match with the regex expression.

### CODE REVIEW

File: tests/falco/events.json [hunk 1]
```diff
@@ -0,0 +1,278 @@
+{
```
Comment (jasondellaluce) [inline tests/falco/events.json:1]: This makes sense! What about putting this under a `tests/data/outputs` directory, and masking this through `run.NewStringFileAccessor`? That would allow the framework to carry around the test file in a runner-agnostic way (it will work whether Falco runs in a container or in the local machine). Accordingly, I would also add the deserialization code in the same package (the part of code responsible of converting the JSON in a Go struct). Take a look at how some of the existing test data is handled.

File: tests/falco/commands_test.go [hunk 2]
```diff
@@ -132,3 +134,38 @@ func TestFalco_Cmd_PluginInfo(t *testing.T) {
...
+	defer evntfile.Close()
+
+	var events Data
+	err = json.NewDecoder(evntfile).Decode(&events)
+	if err != nil {
+		panic(err)
+	}
+
+	runner := tests.NewFalcoExecutableRunner(t)
+	t.Run("all-events-ignored-by-default", func(t *testing.T) {
```
Comment (jasondellaluce) [inline tests/falco/commands_test.go:157]: Since we don't have other tests here, I would avoid having a sub-test with `t.Run` and would just proceed with the test code in the main `TestFalco_Print_IgnoredEvents` test body, what do you think?

File: tests/falco/commands_test.go [hunk 3]
```diff
@@ -132,3 +134,38 @@ func TestFalco_Cmd_PluginInfo(t *testing.T) {
...
+	if err != nil {
+		panic(err)
+	}
+	defer evntfile.Close()
+
+	var events Data
+	err = json.NewDecoder(evntfile).Decode(&events)
+	if err != nil {
+		panic(err)
+	}
```
Comment (jasondellaluce) [inline tests/falco/commands_test.go:154]: Look at the comment below: I think this should be part of the test data package under `tests/data/outputs`, so that it could be accessible by other tests as well and reused.

File: tests/falco/commands_test.go [hunk 4]
```diff
@@ -132,3 +134,38 @@ func TestFalco_Cmd_PluginInfo(t *testing.T) {
...
+	t.Run("all-events-ignored-by-default", func(t *testing.T) {
+		res := falco.Test(
+			runner,
+			falco.WithArgs("-i"),
+		)
+		assert.Regexp(t, regexp.MustCompile(
+			`Ignored\sEvent\(s\):\n`,
+		), res.Stdout())
+		for _, event := range events.Events {
+			assert.Regexp(t, regexp.MustCompile(`\b`+event+`\b`), res.Stdout())
```
Comment (jasondellaluce) [inline tests/falco/commands_test.go:166]: Maybe `assert.Contains` would be more lightweight here, since events are simple known strings. Remember, the more we make tests optimized, the less the CI will take to run them!

File: tests/falco/events.json [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+  "events": [
```
Comment (jasondellaluce) [inline tests/falco/events.json:2]: Out of curiosity, which Falco version did you use to generate this list?

**Review by jasondellaluce — COMMENTED** [review]
Thanks a ton for your first contribution! Much appreciated, specially in this repo!!

**Comment by Rohith-Raju** [comment]
Hi, @jasondellaluce thanks for reviewing my pr. I'll make the requested changes by Saturday as I'm caught up with college exams until tomorrow 😄 
.

File: tests/falco/events.json (hunk 1 above)
Comment (Rohith-Raju) [reply tests/falco/events.json:1]: Sure, will do

File: tests/falco/events.json (hunk 5 above)
Comment (Rohith-Raju) [reply tests/falco/events.json:2]:
```
Falco version: 0.34.1
Libs version:  0.10.4
Plugin API:    2.0.0
Engine:        16
Driver:
  API version:    3.0.0
  Schema version: 2.0.0
  Default driver: 4.0.0+driver
```
Falco version: 0.34.1

File: tests/falco/commands_test.go (hunk 2 above)
Comment (Rohith-Raju) [reply tests/falco/commands_test.go]: >Test the behaviour and expected output of the Falco CLI depending on its options, considered both individually and in combinations:

Yes, since there no combinations for `-i` we can proceed with the main `TestFalco_Print_IgnoredEvents`

File: tests/data/outputs/event.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	"strings"
+
+	"github.com/falcosecurity/testing/pkg/run"
+)
+
+type Data struct {
+	Events []string `json:"events"`
+}
+
+func deserialize() string {
```
Comment (jasondellaluce) [inline tests/data/outputs/event.go:17]: Since we have a fairly recent version of Go, let me suggest considering go:embed https://pkg.go.dev/embed. I think it's a better approach than going through `runtime.Caller` and reinvent the wheel in this case! What do you think?

Plus, if the only point of having a JSON file is to deserialize it and encode the inner list with comma-separated concatenation, why not just encoding all strings as comma-separated in a text file right away? This would also save you the trouble of defining a `Data` struct. I see the attempt of making something reusable, but I would suggest to 1) tackle reusability later when needed and keep the scope of this PR smaller, and 2) the JSON encoding is arbitrary, so it would be less intuitive for future contributors to reproduce.

**TL;DR**: Great work! My last suggestions are:
1. Encode the file as a comma-separated text file containing the list of event names
2. Include it in the project using go:embed, and then making it available through the `NewStringFileAccessor` construct like you already do.

Comment (Rohith-Raju) [reply tests/data/outputs/event.go:17]: >Since we have a fairly recent version of Go, let me suggest considering go:embed https://pkg.go.dev/embed. I think it's a better approach than going through runtime.Caller and reinvent the wheel in this case! What do you think?

Oh, wow I didn't know something like this existed :open_mouth: . Yes, this is great.

>1. Encode the file as a comma-separated text file containing the list of event names
>2. Include it in the project using go:embed, and then making it available through the NewStringFileAccessor construct like you already do.

Will do!!

**Comment by Rohith-Raju** [comment]
Hey @jasondellaluce, So I `exec` ed into the  `falcosecurity/falco-no-driver:master` and ran `falco -i` and it gave me these
```
root@933f5c41b8e2:/# falco -i
2023-04-04T14:57:26+0000: Falco version: 0.34.1-125+2b29ff7 (x86_64)
2023-04-04T14:57:26+0000: Falco initialized with configuration file: /etc/falco/falco.yaml
Ignored I/O syscall(s):
- recv
- pwrite64
- pread64
- sendfile64
- read
- write
- writev
- pwritev
- readv
- sendfile
- preadv
- sendto
- recvfrom
- send
- sendmsg
- recvmsg
- sendmmsg
- recvmmsg
```  
I took these and re-generated the event file. I wasn't able to build `falco` master locally, so the tests are still using the old falco (which the tests are not passing) version and hence failing. Is there a way I can confirm my tests locally? I would love to know your local dev setup

**Comment by jasondellaluce** [comment]
What error are you encountering when compiling Falco? Either way, you can install locally the latest dev version by downloading it from https://download.falco.org/?prefix=packages/bin-dev/. Then, once installed, the test suite should run successfully by following the readme instructions.

Either way, the tests are now just failing in the CI because your regex must be updated to match `Ignored I/O syscall(s):` instead of `Ignored Events:`. The restricted looks go to me, because from v0.35 onwards Falco will only effectively ignore IO-related system calls, whereas all the others will be activated dynamically based on the rulesets and config.

**Comment by Rohith-Raju** [comment]
Oh, I see that there was a new image 14h ago. let me retry

File: tests/data/outputs/events.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+recv,pwrite64,pread64,sendfile64,read,write,writev,pwritev,readv,sendfile,preadv,sendto,recvfrom,send,sendmsg,recvmsg,sendmmsg,recvmmsg
```
Comment (jasondellaluce) [inline tests/data/outputs/events.txt]:
```suggestion
sendfile,pwritev,preadv,writev,read,pwrite,write,sendto,recv,send,sendmsg,recvfrom,recvmsg,pread,sendmmsg,recvmmsg,readv
```
The latest Falco dev fixed a couple of things, this should be the most up to date and hopefully definitive list.

File: tests/data/outputs/events.txt [hunk 8]
```diff
@@ -1 +1 @@
-recv,pwrite64,pread64,sendfile64,read,write,writev,pwritev,readv,sendfile,preadv,sendto,recvfrom,send,sendmsg,recvmsg,sendmmsg,recvmmsg
\ No newline at end of file
+sendfile,recvfrom,readv,sendto,send,read,recvmmsg,write,recvmsg,pwrite,sendmmsg,sendmsg,pread,writev,recv,pwritev,prea
```
Comment (Rohith-Raju) [inline tests/data/outputs/events.txt:1]: @jasondellaluce my latest commit has those changes

File: tests/data/outputs/events.txt [hunk 9]
```diff
@@ -0,0 +1 @@
+sendfile,recvfrom,readv,sendto,send,read,recvmmsg,write,recvmsg,pwrite,sendmmsg,sendmsg,pread,writev,recv,pwritev,preadv
```
Comment (Rohith-Raju) [inline tests/data/outputs/events.txt:1]: @jasondellaluce my list is matching with the latest list.

Comment (Rohith-Raju) [reply tests/data/outputs/events.txt:1]: >What error are you encountering when compiling Falco? Either way, you can install locally the latest dev version by downloading it from https://download.falco.org/?prefix=packages/bin-dev/. Then, once installed, the test suite should run successfully by following the readme instructions.

There was some deps issue but now I've solved it. Used the latest(master) `falco` for the test  and it has since been passing.

**Review by jasondellaluce — APPROVED** [review]
/approve

Thanks a lot and congrats for your first contribution here 🎉
